src/plots/plots.js

```diff
@@ -1352,21 +1368,37 @@ function calculateReservedMargins(margins) {
 }
 
 plots.supplyLayoutModuleDefaults = function(layoutIn, layoutOut, fullData, transitionData) {
+    var components = Object.keys(Registry.componentsRegistry);
+    var basePlotModules = layoutOut._basePlotModules;
     var i, _module;
 
-    // can't be be part of basePlotModules loop
-    // in order to handle the orphan axes case
-    Plotly.Axes.supplyLayoutDefaults(layoutIn, layoutOut, fullData);
+    var Cartesian = Registry.subplotsRegistry.cartesian;
+
+    // check if any components need to add more base plot modules
+    // that weren't captured by traces
+    for(i = 0; i < components.length; i++) {
+        _module = Registry.componentsRegistry[components[i]];
+
+        if(_module.includeBasePlot) {
+            _module.includeBasePlot(layoutIn, layoutOut);
+        }
+    }
+
+    // make sure we *at least* have some cartesian axes
+    if(!basePlotModules.length) {
+        basePlotModules.push(Cartesian);
```

Comment:
In v2, I would like to remove `scatter` (and thus the cartesian base module) from the core bundle so that e.g. gl3d-only custom bundles don't have to require `Cartesian` for nothing. 

In this case, should empty graphs (i.e. with no `basePlotModules`) remain blank (i.e. with no cartesian axes)?

Reply:
Perhaps we just take the first `basePlotModule` and make a subplot from it? Or maybe that's too complicated and we only care about empty plots for contexts like the editor, so if there's no `cartesian` we just leave it blank. Either way shouldn't be too hard, there's not that much code involved in the empty fallback.